src/Compilers/VisualBasic/Portable/Compilation/NamespaceScopeBuilder.vb

```diff
@@ -12,28 +12,29 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
         Public Shared Function BuildNamespaceScope(
             moduleBuilder As Emit.PEModuleBuilder,
-            xmlNamespaces As Dictionary(Of String, XmlNamespaceAndImportsClausePosition),
-            aliasImports As IEnumerable(Of AliasAndImportsClausePosition),
+            xmlNamespacesOpt As IReadOnlyDictionary(Of String, XmlNamespaceAndImportsClausePosition),
+            aliasImportsOpt As IEnumerable(Of AliasAndImportsClausePosition),
             memberImports As ImmutableArray(Of NamespaceOrTypeAndImportsClausePosition),
             diagnostics As DiagnosticBag
         ) As ImmutableArray(Of Cci.UsedNamespaceOrType)
             Dim scopeBuilder = ArrayBuilder(Of Cci.UsedNamespaceOrType).GetInstance
 
             ' first come xml imports
-            If xmlNamespaces IsNot Nothing Then
-                For Each xmlImport In xmlNamespaces
+            If xmlNamespacesOpt IsNot Nothing Then
+                For Each xmlImport In xmlNamespacesOpt
                     scopeBuilder.Add(Cci.UsedNamespaceOrType.CreateXmlNamespace(xmlImport.Key, xmlImport.Value.XmlNamespace))
                 Next
             End If
 
             ' then come alias imports
-            If aliasImports IsNot Nothing Then
-                For Each aliasImport In aliasImports
+            If aliasImportsOpt IsNot Nothing Then
+                For Each aliasImport In aliasImportsOpt
                     Dim target = aliasImport.Alias.Target
                     If target.IsNamespace Then
                         scopeBuilder.Add(Cci.UsedNamespaceOrType.CreateNamespace(DirectCast(target, NamespaceSymbol), aliasOpt:=aliasImport.Alias.Name))
-                    ElseIf Not target.ContainingAssembly.IsLinked
-                        ' We skip alias imports of embedded types to avoid breaking existing code that
+                    ElseIf target.Kind <> SymbolKind.ErrorType AndAlso Not target.ContainingAssembly.IsLinked Then
```

Comment:
Yes, for namespaces it can be null if the namespace is the merged one, but namespaces are handled on the line above.